specification/securityinsights/resource-manager/Microsoft.SecurityInsights/stable/2021-04-01/ThreatIntelligence.json

```diff
@@ -0,0 +1,1183 @@
+          "description": "Source of a threat intelligence entity",
+          "type": "string"
+        },
+        "displayName": {
+          "description": "Display name of a threat intelligence entity",
+          "type": "string"
+        },
+        "description": {
+          "description": "Description of a threat intelligence entity",
+          "type": "string"
+        },
+        "indicatorTypes": {
+          "description": "Indicator types of threat intelligence entities",
+          "items": {
+            "description": "Indicator type of a threat intelligence entity",
+            "type": "string"
+          },
+          "type": "array"
+        },
+        "pattern": {
+          "description": "Pattern of a threat intelligence entity",
+          "type": "string"
+        },
+        "patternType": {
+          "description": "Pattern type of a threat intelligence entity",
+          "type": "string"
+        },
+        "patternVersion": {
+          "description": "Pattern version of a threat intelligence entity",
+          "type": "string"
+        },
+        "killChainPhases": {
+          "description": "Kill chain phases",
+          "items": {
+            "description": "Kill chain phase",
+            "$ref": "#/definitions/ThreatIntelligenceKillChainPhase"
+          },
+          "type": "array"
+        },
+        "parsedPattern": {
+          "description": "Parsed patterns",
+          "items": {
+            "description": "Parsed pattern",
+            "$ref": "#/definitions/ThreatIntelligenceParsedPattern"
+          },
+          "type": "array"
+        },
+        "externalId": {
+          "description": "External ID of threat intelligence entity",
+          "type": "string"
+        },
+        "createdByRef": {
+          "description": "Created by reference of threat intelligence entity",
+          "type": "string"
+        },
+        "defanged": {
+          "description": "Is threat intelligence entity defanged",
+          "type": "boolean"
+        },
+        "externalLastUpdatedTimeUtc": {
+          "description": "External last updated time in UTC",
+          "type": "string"
+        },
+        "externalReferences": {
+          "description": "External References",
+          "items": {
+            "description": "external_reference",
+            "$ref": "#/definitions/ThreatIntelligenceExternalReference"
+          },
+          "type": "array"
+        },
+        "granularMarkings": {
+          "description": "Granular Markings",
+          "items": {
+            "description": "Granular marking",
+            "$ref": "#/definitions/ThreatIntelligenceGranularMarkingModel"
+          },
+          "type": "array"
+        },
+        "labels": {
+          "description": "Labels  of threat intelligence entity",
+          "items": {
+            "description": "label",
+            "type": "string"
+          },
+          "type": "array"
+        },
+        "revoked": {
```

Comment:
>revoked [](http://example.com/codeflow?start=9&length=7)

Generally booleans are discouraged in favor of enums. This would better be modeled as "revocationStatus": enum ["revoked", "unrevoked"].

More details here: [https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html?q=boolean](https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html?q=boolean)

Reply:
Previously reviewed.